com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Utility/HDRP/Water/EvaluateWaterFoamData.cs

```diff
@@ -0,0 +1,69 @@
+using Usage = UnityEditor.ShaderGraph.GraphDelta.GraphType.Usage;
+
+namespace UnityEditor.ShaderGraph.Defs
+    {
+        internal class EvaluateWaterFoamDataNode : IStandardNode
+        {
+            public static string Name => "EvaluateWaterFoamData";
+            public static int Version => 1;
+
+            public static FunctionDescriptor FunctionDescriptor => new(
+                Name,
+    @"FoamData foamData;
+ZERO_INITIALIZE(FoamData, foamData);
+EvaluateFoamData(SurfaceGradient, LowFrequencySurfaceGradient, SimulationFoam, customFoam, UV0.xyz, foamData);
+Smoothness = foamData.smoothness;
+
+",
+                new ParameterDescriptor[]
+                {
+                new ParameterDescriptor("UV0", TYPE.Vec4, Usage.Local, REF.UV0),
+                new ParameterDescriptor("SurfaceGradient", TYPE.Vec3, Usage.In),
+                new ParameterDescriptor("LowFrequencySurfaceGradient", TYPE.Vec3, Usage.In),
+                new ParameterDescriptor("SimulationFoam", TYPE.Float, Usage.In),
+                new ParameterDescriptor("CustomFoam", TYPE.Float, Usage.In),
+                new ParameterDescriptor("SurfaceGradient", TYPE.Vec3, Usage.Out),
```

Comment:
I wonder if it's a problem that there are two parameters named SurfaceGradient? The first one doesn't show up on the node, so I bet it doesn't work.